quantum/aqc/ir/DWFunction.hpp

```diff
@@ -100,11 +100,21 @@ const int nRequiredBits() const override {
    * @param instruction
    */
   void addInstruction(InstPtr instruction) override {
+    xacc::InstructionParameter param = instruction->getParameter(0);
+    bool dupParam = false;
+    for (auto p : parameters) {
+        if (p.as<std::string>() == param.as<std::string>()) {
```

Comment:
I think this will fail at some point - if param is a concrete double for example. You should use toString()